frontend/public/components/utils/webhooks.tsx

```diff
@@ -55,10 +55,51 @@ export const WebhookTriggers: React.FC<WebhookTriggersProps> = (props) => {
     namespace,
   });
   const tableColumnClasses = getTableColumnClasses(canGetSecret);
-  const webhooks = _.filter(triggers, (trigger) => webhookTriggers.has(trigger.type));
+  const [webhookSecrets, setWebhookSecrets] = React.useState<K8sResourceKind[]>([]);
+  const [secretErrors, setSecretErrors] = React.useState<string[]>([]);
+  const [isLoaded, setIsLoaded] = React.useState(false);
+
+  const webhooks: WebhookTrigger[] = _.filter(triggers, (trigger) =>
+    webhookTriggers.has(trigger.type),
+  );
+  const webhookNames: string[] = webhooks.map(
+    (webhook: WebhookTrigger): string => {
+      const triggerProperty = getTriggerProperty(webhook);
+      return _.get(webhook, [triggerProperty, 'secretReference', 'name']);
+    },
+  );
+
+  React.useEffect(() => {
+    if (canGetSecret) {
+      let errors = [];
+      Promise.all(
+        webhookNames.map(
+          (webhookName: string): Promise<K8sResourceKind> => {
+            return k8sGet(SecretModel, webhookName, namespace).then(
+              (secret) => secret,
+              (error) => {
+                errors = _.concat(errors, error.message);
```

Comment:
```suggestion
              ({message = `Could not load secret ${webhookName}.`}) => {
                errors = [...errors, message];
```

Reply:
Would rather keep the error message, since it could differ

Reply:
> Would rather keep the error message, since it could differ

We are keeping the error message, just using es6 syntax instead of `_.concat`